_utils/compliance/validators/connectors_json.py

```diff
@@ -0,0 +1,91 @@
+import argparse
+import json
+from pathlib import Path
+
+from jsonschema import Draft7Validator
+from jsonschema.exceptions import SchemaError
+
+from .base import Validator
+from .models import CheckError, CheckResult
+
+
+class ConnectorsJSONValidator(Validator):
+    @classmethod
+    def validate(cls, result: CheckResult, args: argparse.Namespace) -> None:
+        if not result.options.get("path"):
+            return
+
+        module_dir: Path = result.options["path"]
+        connector_jsons = module_dir.glob("connector_*.json")
+
+        result.options["connectors"] = {}
+        for path in connector_jsons:
+            cls.validate_connector_json(path, result, args)
+
+    @classmethod
+    def validate_connector_json(
+        cls, path: Path, result: CheckResult, args: argparse.Namespace
+    ):
+        try:
+            with open(path, "rt") as file:
+                raw = json.load(file)
+
+        except ValueError:
+            result.errors.append(CheckError(filepath=path, error=f"can't load JSON"))
+            return
+
+        # @todo test for unique identifier
+        if not isinstance(raw.get("name"), str):
+            result.errors.append(
+                CheckError(filepath=path, error=f"`name` is not present")
+            )
+
+        if not isinstance(raw.get("uuid"), str):
+            # @todo possible to fix: generate uuid
+            result.errors.append(
+                CheckError(filepath=path, error=f"`uuid` is not present")
+            )
+        else:
+            if "uuid_to_check" not in result.options:
+                result.options["uuid_to_check"] = {}
+            if path.name not in result.options["uuid_to_check"]:
+                result.options["uuid_to_check"][path.name] = {}
+            result.options["uuid_to_check"][path.name] = raw.get("uuid")
```

Comment:
I wasn't able to validate this check. I modified two modules to have the same ids for two different connectors and not errors were raised.
Even two connectors, in the same module, with the same ids didn't raised errors.